src/main/resources/org/jenkinsci/plugins/badge/BadgeAction/index.groovy

```diff
@@ -30,44 +30,79 @@ l.layout {
 </style>
 """)
 
-        def base =  "${app.rootUrl}${my.project.url}";
-        def badge = base + "badge/icon"
-        
         def fullJobName = h.escape(my.project.fullName);
-        def publicbadge = "${app.rootUrl}buildStatus/icon?job=${fullJobName}";
+        def jobUrlWithView =  "${app.rootUrl}${my.project.url}";
+        def jobUrlWithoutView =  "${app.rootUrl}job/${fullJobName}";
```

Comment:
Real URL to a job inside a folder: `/job/Foldername/job/jobname`
Your code produces: `/job/Foldername/jobname`
This is obviously wrong. Just install Cloudbees Folder Plugin and see how this works.